fmn/rules/utils.py

```diff
@@ -79,10 +79,49 @@ def get_packagers_of_package(config, package):
         _cache.configure(**config['fmn.rules.cache'].copy())
 
     key = cache_key_generator(get_packagers_of_package, package)
-    creator = lambda: _get_pkgdb2_packagers_for(config, package)
+    creator = lambda: _get_packagers_for(config, package)
     return _cache.get_or_create(key, creator)
 
 
+def _get_packagers_for(config, package):
+    if config['fmn.rules.utils.use_pagure_for_ownership']:
+        return _get_pagure_packagers_for(config, package)
+    else:
+        return _get_pkgdb2_packagers_for(config, package)
+
+
+def _get_pagure_packagers_for(config, package):
+    log.debug("Requesting pagure packagers of package %r" % package)
+    default = 'https://src.fedoraproject.org/pagure/api'
+    base = config.get('fmn.rules.utils.pagure_api_url', default)
+
+    # XXX. give a default namespace if one is not provided
+    # Later, we need to come back in here and make all of this namespace-aware.
+    if '/' not in package:
+        package = 'rpms/' + package
+
+    url = '{0}/0/{1}'.format(base, package)
+    log.info("hitting url: %r" % url)
+    response = requests.get(url)
+
+    if not response.status_code == 200:
+        log.warn('URL %s returned code %s', response.url, response.status_code)
+        return set()
+
+    data = response.json()
+
+    packagers = set(sum(data['access_users'].values(), []))
+    groups = set(sum(data['access_groups'].values(), []))
+
+    # XXX - Should we get group membership from pagure instead of FAS?  Ask pingou.
```

Comment:
They can stayed logged in for quite a while, they'll need to relogin for every access to the settings page of a project made 20 minutes or more from their login time, otherwise, their session lasts as long as flask's default which is iirc 31 days.